### ISSUE TITLE
Modify 'aria-labelledby' and `explicit-label` checks' fail messages

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#1004]
## 1. Check: aria-labelledby

- ### Old message
**Fail**: aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty or not visible

- ### Proposed message
**Fail**: aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty

- **Reason:** aria-labelledby referenced to non visible elements is still announced by screen readers

- **Link to demo:** https://codepen.io/iamrafan/pen/ZjGPVe?editors=1000

- **AT Tests** when aria-labelledby used on the interactive elements is referenced to the following spans

  |   | ```<span style="display: none">``` | ```<span style="visibility:   hidden">``` | ```<span   hidden>``` | ```<span   aria-hidden="true">``` | ```<span   role="presentation">```
-- | -- | -- | -- | -- | -- | --
  | **Browser/AT** | **Label   announced?** | **Label   announced?** | **Label   announced?** | **Label   announced?** | **Label   announced?**
1 | **Edge/Narrator** | Yes | Yes | Yes | Yes | No
2 | **Edge/JAWS** | Yes | Yes | Yes | Yes | No
3 | **Edge/NVDA** | Yes | Yes | Yes | Yes | No
4 | **Chrome/NVDA** | Yes | Yes | Yes | Yes | Yes
5 | **Chrome/JAWS** | Yes | Yes | Yes | Yes | Yes
6 | **IE/JAWS** | Yes | Yes | Yes | Yes | Yes
7 | **IE/NVDA** | Yes | Yes | Yes | Yes | Yes
8 | **Safari/Voiceover** | Yes | Yes | Yes | Yes | Yes
9 | **Firefox/NVDA** | Yes | Yes | Yes | Yes | Yes

## 2. Check: explicit-label

- ### Old message
**Fail**: Form element does not have an explicit ```<label>```

- ### Proposed message
**Fail**: Form element does not have an explicit ```<label>``` or has an explicit ```<label>``` that is not accessible (```<label>``` has ‘hidden’ or ‘aria-hidden’ attribute OR has CSS styling with ‘display: none’ or ‘visibility: hidden’)

- **Reason:** Explicit labels with the above properties are not announced by screen readers

- **Link to demo:** https://codepen.io/iamrafan/pen/gjpybY?editors=1000

- **AT tests** 

  |   | ```<label for="element-1"   style="display:   none">``` | ```<label for="element-1"   style="visibility: hidden">``` | ```<label   for="element-1" hidden>``` | ```<label   for="element-1" aria-hidden="true">``` | ```<label   for="element-1" role="presentation">```
-- | -- | -- | -- | -- | -- | --
  | **Browser/AT** | **Label announced?** | **Label announced?** | **Label announced?** | **Label announced?** | **Label announced?** 
1 | **Edge/Narrator** | No | No | No | No | Yes
2 | **Edge/JAWS** | No | No | No | No | Yes
3 | **Edge/NVDA** | No | No | No | No | Yes
4 | **Chrome/NVDA** | No | No | No | Yes | Yes
5 | **Chrome/JAWS** | No | No | No | Yes | Yes
6 | **IE/JAWS** | Yes | Yes | Yes | Yes | Yes
7 | **IE/NVDA** | Yes | Yes | Yes | Yes | Yes
8 | **Safari/voiceover** | No | No | No | No | No
9 | **Firefox/NVDA** | No | No | No | Yes | No

<pre><code>axe-core version: 3.0.3
axe-Coconut version: 3.4.0-alpha.2</code></pre></code></pre>

**Comment by WilcoFiers** [comment]
You are darn efficient getting screen reader testing done. How do you pull that off?

Also +1 to the fix. I'll sort it out right away.

**Comment by iamrafan** [comment]
I got used to it :) . Thanks for the fix. 
Could you please look into the explicit-label message as well

**Comment by WilcoFiers** [comment]
I think @marcysutton was working on something for explicit label messages - I thought we were going to add a check? I can't recall, Marcy, where were we with this?

**Comment by marcysutton** [comment]
@WilcoFiers you're probably thinking of #242, which got merged? https://github.com/dequelabs/axe-core/pull/792

**Comment by WilcoFiers** [comment]
Jup, so I think we resolved this. @iamrafan does 792 look alright to you? Feel free to raise a new issue if there's anything else.
